src/SFA.DAS.ApprenticeCommitments.Web/Identity/RequiresIdentityConfirmed.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using SFA.DAS.ApprenticeCommitments.Web.Services;
+
+#nullable enable
+
+namespace SFA.DAS.ApprenticeCommitments.Web.Identity
+{
+    public class RequiresIdentityConfirmedAttribute : TypeFilterAttribute
+    {
+        public RequiresIdentityConfirmedAttribute() : base(typeof(ClaimRequirementFilter))
+        {
+        }
+    }
+
+    public class ClaimRequirementFilter : IAuthorizationFilter
+    {
+        public void OnAuthorization(AuthorizationFilterContext context)
+        {
+            if (VerifiedUser.UserHasUnconfirmedIdentity(context.HttpContext))
+                context.Result = new RedirectResult("/ConfirmYourIdentity");
+        }
```

Comment:
Personally I think this should simply throw an exception, but redirecting at this point has it's merits

Reply:
The thing I like about this way is that we don't need code in another class somewhere else to catch that exception and set the redirect - everything is contained in these 30 lines right here.

Reply:
yes there are definitely merits in doing it this way